### PR TITLE
Add mish and softplus activation function

### PR DESCRIPTION
**Pull request by pacowong**
### Checklist

- [x] Confirmed that `run-checks all` script has been executed.
- [x] Made sure the book is up to date with changes in this PR.

### Related Issues/PRs
NA

### Changes
Mish demonstrated better results than both Swish and ReLU in [1]. It has been included in PyTorch as well.

[1]: https://arxiv.org/abs/1908.08681

### Testing
Please refer to the new test cases.

### CODE REVIEW

**Review by antimora — COMMENTED** [review]
Thanks for addition. Can you also please add new operators in the book too?

**Review by louisfd — CHANGES_REQUESTED** [review]
I verified the equations, everything looks good to me. 
It will be good to go once the book is updated and CI is fixed (you seem to have a formatting problem, run `cargo fmt --all`)

**Comment by pacowong** [comment]
Thanks @antimora and @louisfd for the comments. The new functions are added to the book. [0aeaa71](https://github.com/tracel-ai/burn/pull/1071/commits/0aeaa7181887cd79469307c4d2b5a2046c5c1be3) and the formatting issue is fixed.

File: burn-dataset/Cargo.toml [hunk 1]
```diff
@@ -18,7 +18,7 @@ audio = ["hound"]
 fake = ["dep:fake"]
 
 sqlite = ["__sqlite-shared", "dep:rusqlite"]
-sqlite-bundled = ["__sqlite-shared", "rusqlite/bundled"]
+sqlite-bundled = ["__sqlite-shared", "rusqlite?/bundled"]
```
Comment (nathanielsimard) [inline burn-dataset/Cargo.toml:21]: I think the `?` is making the tests fail, and it's a bit unnecessary in this case.

**Review by nathanielsimard — APPROVED** [review]
Thanks for the addition; the only remaining work is to fix the CI. You can run it locally to debug things to help you, but I think it's related to my comment below.

File: burn-dataset/Cargo.toml (hunk 1 above)
Comment (pacowong) [reply burn-dataset/Cargo.toml]: Thanks. Not sure if my Rust is too new.

According to the [documentation] (https://doc.rust-lang.org/cargo/reference/features.html), 
```
You can add a ? as in "package-name?/feature-name" which will only enable the given feature if something else enables the optional dependency.
```

Comment (nathanielsimard) [reply burn-dataset/Cargo.toml]: No Rust isn't too new, it's just that `rusqlite` is necessary for the feature flag to work I think.

**Pull request merged**